build.rs

```diff
@@ -0,0 +1,12 @@
+fn main() {
+    // prefer method A if both method A and B are selected
+    if cfg!(feature = "__aarch64") || cfg!(target_arch = "aarch64") {
```

Comment:
I don't understand why we need separate features for arm and riscv in addition to the existing target_arch config.

Reply:
I added these for easier compile-checking. This means I can just do `cargo check --feature=__aarch64` to check if the code would compile on aarch64 without installing a target toolchain.